src/main/java/org/opensearch/flowframework/workflow/RegisterLocalModelStep.java

```diff
@@ -0,0 +1,190 @@
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        modelName = (String) content.get(NAME_FIELD);
+                        break;
+                    case VERSION_FIELD:
+                        modelVersion = (String) content.get(VERSION_FIELD);
+                        break;
+                    case DESCRIPTION_FIELD:
+                        description = (String) content.get(DESCRIPTION_FIELD);
+                        break;
+                    case MODEL_FORMAT:
+                        modelFormat = MLModelFormat.from((String) content.get(MODEL_FORMAT));
+                        break;
+                    case MODEL_GROUP_ID:
+                        modelGroupId = (String) content.get(MODEL_GROUP_ID);
+                        break;
+                    case MODEL_TYPE:
+                        modelType = (String) content.get(MODEL_TYPE);
+                        break;
+                    case EMBEDDING_DIMENSION:
+                        embeddingDimension = (String) content.get(EMBEDDING_DIMENSION);
+                        break;
+                    case FRAMEWORK_TYPE:
+                        frameworkType = FrameworkType.from((String) content.get(FRAMEWORK_TYPE));
+                        break;
+                    case ALL_CONFIG:
+                        allConfig = (String) content.get(ALL_CONFIG);
+                        break;
+                    case MODEL_CONTENT_HASH_VALUE:
+                        modelContentHashValue = (String) content.get(MODEL_CONTENT_HASH_VALUE);
+                        break;
+                    case URL:
+                        url = (String) content.get(URL);
+                        break;
+                    default:
+                        break;
+
+                }
+            }
+        }
+
+        if (Stream.of(modelName, modelVersion, modelFormat, modelGroupId, embeddingDimension, frameworkType, modelContentHashValue, url)
+            .allMatch(x -> x != null)) {
+
+            // Create model configuration, assuming null pooling mode, null model max length, normalize results set to false
+            TextEmbeddingModelConfigBuilder modelConfigBuilder = TextEmbeddingModelConfig.builder()
+                .modelType(modelType)
+                .embeddingDimension(Integer.valueOf(embeddingDimension))
+                .frameworkType(frameworkType)
+                .poolingMode(null)
+                .modelMaxLength(null)
+                .normalizeResult(false);
+
+            if (allConfig != null) {
+                modelConfigBuilder.allConfig(allConfig);
+            }
+
+            MLModelConfig modelConfig = modelConfigBuilder.build();
+
+            MLRegisterModelInput mlInput = MLRegisterModelInput.builder()
+                .modelName(modelName)
```

Comment:
We are not adding `description` field here. It can be optional but need to be included if present. For handling optional params refer [this](https://github.com/opensearch-project/opensearch-ai-flow-framework/blob/main/src/main/java/org/opensearch/flowframework/workflow/ModelGroupStep.java#L118-L132)

Reply:
ah good catch, I have added this to both register remote model and register local model